src/libraries/System.Text.Json/src/System/Text/Json/TypeExtensions.cs

```diff
@@ -7,31 +7,29 @@ namespace System.Text.Json
 {
     internal static class TypeExtensions
     {
+        private static readonly Type s_NullableType = typeof(Nullable<>);
+
         /// <summary>
         /// Returns <see langword="true" /> when the given type is of type <see cref="Nullable{T}"/>.
         /// </summary>
-        public static bool IsNullableValueType(this Type type)
-        {
-            return Nullable.GetUnderlyingType(type) != null;
-        }
+        public static bool IsNullableOfT(this Type type) =>
+            type.IsGenericType && type.GetGenericTypeDefinition() == s_NullableType;
 
         /// <summary>
         /// Returns <see langword="true" /> when the given type is either a reference type or of type <see cref="Nullable{T}"/>.
         /// </summary>
-        public static bool IsNullableType(this Type type)
-        {
-            return !type.IsValueType || IsNullableValueType(type);
-        }
+        /// <remarks>This calls <see cref="Type.IsValueType"/> which is slow. If knowledge already exists
+        /// that the type is a value type, call <see cref="IsNullableOfT"/> instead.</remarks>
+        public static bool CanBeNull(this Type type) =>
+            !type.IsValueType || (type.IsGenericType && type.GetGenericTypeDefinition() == s_NullableType);
```

Comment:
This could be `!type.IsValueType || type.IsNullableOfT()` but kept it verbose to avoid the potential overhead of calling another method.

Reply:
> potential overhead of calling another method.

Is this wrt to running on mono interpreter, similar to changes in https://github.com/dotnet/runtime/pull/39323/files#diff-56743101af5d377fb408240c8ec57603R17-R18?

Reply:
Nope. Instead it is attempt at manual inlining. Here's the four possible configurations:
1) Call `IsNullableOfT()`. Slowest
2) Manual inlining (as above). 4% faster than (1)
3) Call `IsNullableOfT()` that has `[AggressiveInlining]`. 1% faster than (2)
4) Same above but add `[AggressiveInlining]` to `CanBeNull`. 1% faster than (3)

Each option above gets progressively faster (with the case of a nullable value type) and the biggest difference was avoiding the method call to `IsNullableOfT()`.

I was trying to avoid `[AggressiveInlining]` but since this on a hot path (called per property in some cases) and the methods are small, I suppose `[AggressiveInlining]` is justified so I'll make the change to go from (2) to (4)